### ISSUE TITLE
Accessibility considerations for the Composer and Autocomplete

### ISSUE THREAD

**Issue by t3chguy** [vector-im/riot-web#11075]
Prior to recent changes the autocomplete has not been whatsoever screen-reader accessible, whilst it is now a lot better, keyboard accessibility of the composer remains a big problem and it seems to compete with its choice of function.

The composer locks the focus to itself, there is no way, using TAB, to get out of it.
There used to a be a feature where TAB on composer without any autocomplete suggestions would complete the latest user having spoken (https://github.com/vector-im/riot-web/issues/4770), but this means that TAB is always hijacked and it is no longer possible to leave the composer. Now that the composer gets automatically focused on any unhandled text input anywhere on the app I think this should be scrapped, and instead `@`TAB result in mentioning the latest user.

Using TAB for navigation of the autocomplete is still different to competitors such as Slack, where TAB selects the completion but this should not ruin accessibility entirely like disallowing TAB when there are no possible completions.

**Comment by t3chguy** [comment]
Further, learning from Slack, they do not put the completion into the composer until it is confirmed, which means that as soon as the completion pane comes up, the first one can be selected and thusly read out by the screen reader as a notion of showing the autocomplete has shown up (https://github.com/vector-im/riot-web/issues/11011)

**Comment by t3chguy** [comment]
Further, learning from Slack, they do not put the completion into the composer until it is confirmed, which means that as soon as the completion pane comes up, the first one can be selected and thusly read out by the screen reader as a notion of showing the autocomplete has shown up (https://github.com/vector-im/riot-web/issues/11011)

**Comment by lampholder** [comment]
We discussed this in the product sync, and decided that an easy, noncontroversial decision was that tabbing in an empty composer should escape you from the composer to navigate around the rest of the app.

Otherwise, people use typing-the-first-three-characters-of-somebody's-name-then-pressing-tab-to-autocomplete-them, which would be _very_ annoying to tab out of the composer if the autocomplete happened not to match a room member, so we weren't able to make a decision to make more invasive changes to composer/autocomplete at this stage.

How does that osund?

**Comment by t3chguy** [comment]
Tabbing in an empty composer is a step forward for sure, but what about someone mid-way through a message who wants to send an image before their message or to refer back to a message in the timeline

**Comment by t3chguy** [comment]
> Well, Riot is the only one with an autocomplete that works like this, and it irritates me each time I use it and have to remember to use Space, where all other autocompletes I know (Pinafore, Twitter, Slack, IRCCloud, Gmail, and others) use tab to autocomplete. 😉 And I think, this mirrors most mainstream, non-Riot-dev, user expectations.

- Marco Zehe

**Comment by t3chguy** [comment]
Closing in favour of https://github.com/vector-im/riot-web/issues/11071
